src/components/VoteButton.jsx

```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react'
+import { voteReview } from '../api.js'
+
+export function VoteButton({ reviewID, setReview, setDisplayAlert }) {
+  function vote(incVote) {
+    const req = { inc_votes: incVote }
+    setReview((review) => {
+      // optimistic rendering
+      const updatedReview = { ...review }
+      updatedReview.votes += incVote
+      return updatedReview
+    })
+    voteReview(reviewID, req)
+      .then(() => {
+        return
+      })
+      .catch(() => {
+        setReview((review) => {
+          // undo optimistic rendering
+          const updatedReview = { ...review }
+          updatedReview.votes -= incVote
+          return updatedReview
+        })
+        setDisplayAlert(true)
```

Comment:
Apologies, but once again I'm not sure what this does as I can't visualise or see it in action on the app. If it is doing something, then fine.